src/libprojectM/Renderer/BeatDetect.hpp

```diff
@@ -32,65 +32,63 @@
 #define _BEAT_DETECT_H
 
 #include "../PCM.hpp"
-#include <algorithm>
-#include <cmath>
-
-
-// this is the size of the buffer used to determine avg levels of the input audio
-// the actual time represented in the history depends on FPS
-#define BEAT_HISTORY_LENGTH 80
+#include <array>
 
 class BeatDetect
 {
-	public:
-        // Does this really belong here? maybe belongs on projectM.Settings?
-        float beatSensitivity;
+public:
+    // Does this really belong here? maybe belongs on projectM.Settings?
+    float beatSensitivity{1.f};
+
+    float treb{0.f};
+    float mid{0.f};
+    float bass{0.f};
+    float vol_old{0.f};
+
+    float treb_att{0.f};
+    float mid_att{0.f};
+    float bass_att{0.f};
+    float vol{0.f};
+    float vol_att{0.f};
 
-        float treb ;
-		float mid ;
-		float bass ;
-		float vol_old ;
+    PCM& pcm;
 
-		float treb_att ;
-		float mid_att ;
-		float bass_att ;
-		float vol;
-        float vol_att ;
+    /** Methods */
```

Comment:
Better place methods on top of member declarations.

Reply:
Done